clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1902,7 +1916,9 @@ private void internalClose(boolean doFlush) {
     // Only reset Offset and Drop Partition Messages are important, subscribe/unsubscribe will be handled
     // on the restart by Helix Controller notifications on the new StoreIngestionTask.
     this.storeRepository.unregisterStoreDataChangedListener(this.storageUtilizationManager);
-    for (ConsumerAction message: consumerActionsQueue) {
+    // Remove all the actions out of the queue in order
+    while (!consumerActionsQueue.isEmpty()) {
+      ConsumerAction message = consumerActionsQueue.poll();
```

Comment:
why we change to remove actions from queue?

Reply:
In theory, it doesn't matter. Since the queue only exists in the SIT instance, if SIT is closed, no one can access to the queue anyway.
IIRC, a test case related to kill job failed, the test tries to verify that there is no active subscription after killing the push, but since we also check subscription action in the queue, the check would fail.